File: Лабораторная работа 1. Шифрование шифрами однозначной замены/NOTINTERFACElab1/Oldref/CipherATBASH/ATBASH.js
function atbashCipher(input) {
    var alphabet = "АБВГДЕЖЗИЙКЛМНОПРСТУФХЦЧШЩЪЫЬЭЮЯ";
    var punctuations = {
        ",": "ЗПТ",
        ".": "ТЧК",
        "-": "ТИРЕ",
    };
    var preprocessedInput = input
        .toUpperCase()
        .replace(/[,.-]/g, function (match) { return punctuations[match]; }); // Преобразовываем знаки препинания
    return preprocessedInput
        .split('')
        .map(function (char) {
        if (alphabet.includes(char)) {
            var index = alphabet.indexOf(char);
            return alphabet.charAt(alphabet.length - index - 1);
        }
        else {
            return char;
        }
    })
        .join('');
}
var originalText = 'Тот, кто ложится на два стула, падает на ребра.';
var preprocessedText = originalText.replace(/[,.-]/g, function (match) { return match === ',' ? 'ЗПТ' : match === '.' ? 'ТЧК' : 'ТИРЕ'; });
var encryptedText = atbashCipher(preprocessedText);
console.log('Оригинальный текст:', originalText);
console.log('Предварительно обработанный текст:', preprocessedText);
console.log('Зашифрованный текст:', encryptedText);

File: Лабораторные работы КРИПТОГРАФИЯ/Шифрование шифрами однозначной замены/It's working/atbash.py
def atbash_encrypt(open_text, alphabet):
    encrypted_text = ""  # Шифртекст
    for element in open_text:  # Проход по всем символам открытого текста
        encrypted_text += alphabet[len(alphabet) - alphabet.index(element) - 1]  # Добавление в итоговый шифртекст зашифрованного символа
    return encrypted_text  # Возврат шифртекста

def atbash_decrypt(encrypted_text, alphabet):
    decrypted_text = ""  # Расшифрованный текст
    for element in encrypted_text:  # Проход по всем символам шифртекста
        decrypted_text += alphabet[len(alphabet) - alphabet.index(element) - 1]  # Добавление в итоговый текст расшифрованного символа
    # Перевод символов из их текстовых значений в символьные
    decrypted_text = decrypted_text.replace("тчк", ".").replace("зпт", ",").replace("тире", "-").replace('прбл', ' ').replace('двтч', ':').replace('тчсзп', ';').replace('отскб', '(').replace('зкскб', ')').replace('впрзн', '?').replace('восклзн', '!').replace('првст', '\n')
    return decrypted_text  # Возврат расшифрованного текста